app/main/helpers/diff_tools.py

```diff
@@ -1,209 +1,132 @@
-                and not include_unchanged_lines_in_output:
-            continue
-
-        lines['revision_1'].append(
-            Markup(render_words_html(revision_1_words, index+1))
-        )
-        lines['revision_2'].append(
-            Markup(render_words_html(revision_2_words, index+1))
-        )
-
-    return lines
-
-
-def split_words_diff(words_diff):
-    """
-    Accepts a list of words that have been diffed
-    Returns two lists: one with removals ('-') and one with additions ('+')
-    Ignores detail ('?') lines
-
-    In:  ['  Hi', '- there', '+ there!', '?      +\n']
-    Out: (['  Hi', '- there'], ['  Hi', '+ there!'])
-    """
-
-    return [w for w in words_diff
-            if get_word_type(w) in ['unchanged', 'removal']], \
-           [w for w in words_diff
-            if get_word_type(w) in ['unchanged', 'addition']]
-
-
-def render_words_html(words, line_number):
-
-    def _render_words_inner_html(words, inferred_type):
-
-        html_words = []
-
-        for word in words:
-            word = escape(word)
-            type = get_word_type(word)
-
-            if type == 'unchanged':
-                html_words.append(u"{}".format(word[2:]))
-
-            elif type == inferred_type:
-                html_words.append(u"<strong>{}</strong>".format(word[2:]))
-
-        return ' '.join(html_words).replace(u'</strong> <strong>', ' ')
-
-    # Can only have one inferred type per line
-    inferred_type = get_line_type(words)
-
-    return \
-        u"<td class='line-number line-number-{type}'>{line_number}</td>" \
-        u"<td class='line-content {type}'>{line}</td>".format(
-            type=inferred_type,
-            line_number=line_number,
-            line=_render_words_inner_html(words, inferred_type)
-        )
-
-
-def get_words_diff(revision_1_line, revision_2_line):
-    """
-    Accepts two strings which are split into lists
-    Returns a list of word-diffs
-
-    In:  'Hi there', 'Hi there!'
-    Out: ['  Hi', '- there', '+ there!', '?      +\n']
-    """
-    differ = difflib.Differ()
-    # Splitting the lines on whitespace to that we get a word diff
-    return list(differ.compare(
-        revision_1_line.split(), revision_2_line.split()
-    ))
+from itertools import chain
 
+from flask import Markup, render_template
+from flask._compat import string_types
 
-def get_line_type(words, types_to_look_for=None):
+from lxml import html
 
-    if not words:
-        return 'empty'
+from dmcontent.questions import Multiquestion
 
-    if not types_to_look_for:
-        types_to_look_for = ['addition', 'removal']
 
-    for word in words:
-        type = get_word_type(word)
-        if type in types_to_look_for:
-            return type
+def _question_iter(question):
+    # type testing is weird - this should eventually belong in the question subclass itself
+    if isinstance(question, Multiquestion):
+        for sub_question in question.questions:
+            yield sub_question
```

Comment:
You will need to recurse here as we have some multiquestions that contain multiquestions.